syntax = "proto3";

package chat;

// Import Timestamp from Protobuf for timestamps
import "google/protobuf/timestamp.proto";

message ChatMessage {
  int32 id = 1;
  string content = 2;
  int32 sender_id = 3;
  int32 chat_room_id = 4;
  google.protobuf.Timestamp timestamp = 5;
}

message ChatRoom {
  int32 id = 1;
  string name = 2;
  google.protobuf.Timestamp created_at = 3;
  int32 game_id = 4;
}

message CreateChatRoomRequest {
  string name = 1;
  int32 game_id = 2;
}

message GetChatRoomRequest {
  int32 id = 1;
}

message Empty {}

service ChatService {
  rpc CreateChatRoom(CreateChatRoomRequest) returns (ChatRoom);
  rpc GetChatRoomById(GetChatRoomRequest) returns (ChatRoom);
  rpc ListChatRooms(Empty) returns (stream ChatRoom);
}